### PR TITLE
Integrated the crop UI into data processing

### PR DESCRIPTION
**Pull request by meriam04**
To process the facial expression data, the `data_processing.py` file has been updated to function as follows: 

1. Provide a path to `video_dirs` where all the emotion-labeled videos collected from the experiment should be located. It expects .mp4 files in this directory.
2. Specify an `output_path` where all the final cropped images should be moved.
3. Specify whether the output should be considered for binary or multiclass classification. The default is binary (true).
4. There is an optional command-line argument for `get_frames` (true by default). If this is set to false, it will not run the `extract_frames` function. This should only be used if the frames have already been extracted and we want to re-crop and change the output path or classification mode.

Also note that the resizing no longer happens here. Instead it is in the train.py file using the tensorflow library, once it loads in the cropped images.

### CODE REVIEW

File: data_processing/face/crop_ui.py [hunk 1]
```diff
@@ -203,25 +222,60 @@ def run_image_cropper():
...
     )
 
     # If an image is selected, run the image cropper application
-    if image_path:
-        # Open the selected image file and get its dimensions
-        img = Image.open(image_path)
-        width, height = img.size
+    if not image_path:
+        logging.error("ImageCropper: no image selected")
+        return
```
[the remark this answers is not in the document]
Comment (meriam04) [reply data_processing/face/crop_ui.py:231]: That makes sense. I have changed it to raise an error.

File: data_processing/face/crop_ui.py [hunk 2]
```diff
@@ -174,24 +179,38 @@ def crop_all_images(self):
...
+def run_image_cropper(initial_dir=os.getcwd()):
+    """
+    Run the image cropper application.
+    NOTE: This function sometimes has bugs when called repeatedly.
+    """
+
+    if not os.path.exists(initial_dir):
+        initial_dir = os.getcwd()
+
+    # cropped_directory = None
```
[the remark this answers is not in the document]
Comment (meriam04) [reply data_processing/face/crop_ui.py:212]: No, forgot to delete it.

File: data_processing/face/crop_ui.py [hunk 3]
```diff
@@ -203,25 +222,60 @@ def run_image_cropper():
...
 
     # If an image is selected, run the image cropper application
-    if image_path:
-        # Open the selected image file and get its dimensions
-        img = Image.open(image_path)
-        width, height = img.size
+    if not image_path:
+        logging.error("ImageCropper: no image selected")
+        return
+
```
[the remark this answers is not in the document]
Comment (meriam04) [reply data_processing/face/crop_ui.py]: Yes, I have changed it so that it no longer repeats code.

**Review by phonycrow — APPROVED** [review]
Everything looks good. Great work!

**Pull request merged**